cmd/monaco/supportarchive/supportarchive.go

```diff
@@ -29,11 +30,39 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/zip"
 )
 
-type SupportArchive struct {
-	Value bool
+type ctxKey struct {
+	value bool
 }
 
-func Archive(fs afero.Fs) error {
+// NewContextWithSupportArchive creates a new child context that has a signal that the support archive is required.
+// If the command has already been called, it returns the given context.
+func NewContextWithSupportArchive(ctx context.Context) context.Context {
+	if ctx.Value(ctxKey{}) != nil {
+		return ctx
+	}
+
+	key := &ctxKey{}
+	return context.WithValue(ctx, ctxKey{}, key)
+}
+
+func IsEnabled(ctx context.Context) bool {
+	v := ctx.Value(ctxKey{})
+	return v != nil && v.(*ctxKey).value
+}
+
+// Enable sets the support archive as enabled in the given context.
+// Prior to this, NewContextWithSupportArchive needs to be called
+func Enable(ctx context.Context) {
+	v := ctx.Value(ctxKey{})
+
+	if v == nil {
+		panic("support archive not created")
```

Comment:
Why? Why not just set the value?

Reply:
We can't just set the value because it would set it in _this_ `context.Context` if it doesnt exist rather than the parent  (or great-great-uhr-alt-old-context), meaning when we go to check if the support archive has been enabled later on, we wont have that information anymore.

Reply:
Context Value is used here in a different way.
At the very beginning of "Monaco" execution, there is a defer function that prints a support archive even if panic occurs. On the other hand, information about whether the archive should be printed is collected later in the program. For that reason, it is a 2-step process (ContextWithSupportArchive and Enable). First needs to happen at the very beginning before the defer function is specified, and second at the point when cobra framework gives us values of flags.
To avoid using global variables (even non-exported), the value in the context is a pointer. With it, we have the backdoor to alter the value later during the execution and get information in the defer function.

Reply:
We discovered the cobra.OnFinalize command, which works well for this use case.